addons/stock/models/stock_location.py

```diff
@@ -261,6 +261,9 @@ def _get_putaway_strategy(self, product, quantity=0, package=None, packaging=Non
                     continue
                 putaway_rules -= putaways
 
+            # Put the priority on putaway with the same package_type
+            putaway_rules.sorted(lambda pa: package_type in pa.package_type_ids or package_type == pa.package_type_ids)
```

Comment:
False < True, and `sorted` is not in-place
`sort(putaway_rules, lambda pa: !(package_type in pa.package_type_ids or package_type == pa.package_type_ids))`